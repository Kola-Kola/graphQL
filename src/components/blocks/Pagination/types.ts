export interface IProps {
  currentPage: number;
  totalElements: number;
  numberOfElements: number;
  onChangePagination: (currentPage: number) => void;
}
