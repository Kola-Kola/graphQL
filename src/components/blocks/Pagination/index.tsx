import { IProps } from "./types";
import { ReactComponent as Previous } from "../../assets/previous.svg";
import { ReactComponent as Next } from "../../assets/next.svg";
import ButtonSquare from "../../controls/ButtonSquare";
import {
  Container,
  PaginationInformations,
  ButtonSquareWithMarginRight,
} from "./styles";

const Pagination = ({
  currentPage,
  totalElements,
  numberOfElements,
  onChangePagination,
}: IProps) => {
  const onChange = (state: string) => {
    const current = state === "previous" ? currentPage - 1 : currentPage + 1;

    if (current < 1) {
      return onChangePagination(1);
    }

    if (current > numberOfElements / 5) {
      onChangePagination(numberOfElements / 5);
    }

    onChangePagination(current);
  };
  return (
    <Container>
      <PaginationInformations data-testid="PaginationInformations">
        {currentPage} â€” {totalElements} of {numberOfElements}
      </PaginationInformations>
      <div>
        {numberOfElements / 5 > 1 && (
          <ButtonSquareWithMarginRight
            data-testid="ButtonSquareWithMarginRight"
            onClick={() => onChange("previous")}
          >
            <Previous />
          </ButtonSquareWithMarginRight>
        )}
        {totalElements === 5 && (
          <ButtonSquare
            data-testid="ButtonSquare"
            onClick={() => onChange("next")}
          >
            <Next />
          </ButtonSquare>
        )}
      </div>
    </Container>
  );
};

export default Pagination;
