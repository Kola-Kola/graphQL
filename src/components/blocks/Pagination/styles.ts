import styled from "styled-components";
import { bodyBook16 } from "../../utility/typography";
import ButtonSquare from "../ButtonSquare";

export const PaginationInformations = styled.p`
  margin: 0;
  ${bodyBook16}
`;

export const ButtonSquareWithMarginRight = styled(ButtonSquare)`
  margin-right: 8px;
`;

export const Container = styled.section`
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
  margin-top: 28px;
  padding: 0 28px;

  @media only screen and (max-width: 760px),
    (min-device-width: 768px) and (max-device-width: 1024px) {
    display: none;
  }
`;
