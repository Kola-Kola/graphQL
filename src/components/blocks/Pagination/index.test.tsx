import { render, screen, fireEvent } from "@testing-library/react"; // (or /dom, /vue, ...)
import Pagination from "./index";
import renderer from "react-test-renderer";
import Theme from "../../layout/Theme";
import "jest-styled-components";

describe("Pagination Component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Theme>
          <Pagination
            currentPage={1}
            totalElements={5}
            numberOfElements={50}
            onChangePagination={() => {}}
          />
        </Theme>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render Pagination with all features", () => {
    render(
      <Theme>
        <Pagination
          currentPage={1}
          totalElements={5}
          numberOfElements={50}
          onChangePagination={() => {}}
        />
      </Theme>
    );

    expect(screen.getByTestId("PaginationInformations")).toBeInTheDocument();
    expect(
      screen.getByTestId("ButtonSquareWithMarginRight")
    ).toBeInTheDocument();
    expect(screen.getByTestId("ButtonSquare")).toBeInTheDocument();

    fireEvent.click(screen.getByTestId("ButtonSquare"));
  });

  it("Should render Pagination with partial features", () => {
    render(
      <Theme>
        <Pagination
          currentPage={1}
          totalElements={5}
          numberOfElements={5}
          onChangePagination={() => {}}
        />
      </Theme>
    );

    expect(screen.getByTestId("PaginationInformations")).toBeInTheDocument();
    expect(
      screen.queryByTestId("ButtonSquareWithMarginRight")
    ).not.toBeInTheDocument();
    expect(screen.getByTestId("ButtonSquare")).toBeInTheDocument();
  });
});
