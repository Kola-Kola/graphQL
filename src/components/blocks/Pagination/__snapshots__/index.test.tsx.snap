// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component renders correctly 1`] = `
.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: #FC5C63;
  border: none;
  border-radius: 4px;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 32px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  vertical-align: middle;
  width: 32px;
}

.c2:focus {
  outline: 1px dashed #FC5C63;
}

.c2:hover {
  background-color: #FC5057;
}

.c1 {
  margin: 0;
  font-family: Maison Neue,sans-serif;
  font-size: 16px;
  line-height: 24px;
  font-weight: 400;
}

.c3 {
  margin-right: 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 28px;
  padding: 0 28px;
}

@media only screen and (max-width:760px), (min-device-width:768px) and (max-device-width:1024px) {
  .c0 {
    display: none;
  }
}

<section
  className="c0"
>
  <p
    className="c1"
    data-testid="PaginationInformations"
  >
    1
     â€” 
    5
     of 
    50
  </p>
  <div>
    <button
      className="c2 c3"
      data-testid="ButtonSquareWithMarginRight"
      onClick={[Function]}
    >
      <svg>
        previous.svg
      </svg>
    </button>
    <button
      className="c2"
      data-testid="ButtonSquare"
      onClick={[Function]}
    >
      <svg>
        next.svg
      </svg>
    </button>
  </div>
</section>
`;
